monai/handlers/classification_saver.py

```diff
@@ -67,7 +75,7 @@ def attach(self, engine: Engine) -> None:
             self.logger = engine.logger
         if not engine.has_event_handler(self, Events.ITERATION_COMPLETED):
             engine.add_event_handler(Events.ITERATION_COMPLETED, self)
-        if not engine.has_event_handler(self.saver.finalize, Events.COMPLETED):
+        if self._expected_rank and not engine.has_event_handler(self.saver.finalize, Events.COMPLETED):
             engine.add_event_handler(Events.COMPLETED, lambda engine: self.saver.finalize())
```

Comment:
Should () be at the end of `self.saver.finalize()`?

Reply:
Hi @ericspod ,

The handler callback registered here is a lambda function: `lambda engine: self.saver.finalize()`, not the `finalize` function itself.

Thanks.

Reply:
Oops I missed that!